apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.ts

```diff
@@ -276,6 +277,42 @@ GoogleBlockly.Extensions.register(
         this.updateEnabled_();
         this.updateParameters_();
       },
+      /**
+       * Returns the data model for this procedure block, or finds it if it has not been set.
+       *
+       * @returns The data model for this procedure block.
+       */
+      getProcedureModel: function (
+        this: ProcedureBlock
+      ): IProcedureModel | null {
+        if (!this.model_) {
+          this.model_ = this.findProcedureModel_(
+            this.getFieldValue('NAME'),
+            this.paramsFromSerializedState_
+          );
+        }
+        return this.model_;
+      },
+
+      /**
+       * Makes sure that if we are updating the parameters before any move events
+       * have happened, the args map records the current state of the block. Does
+       * not remove entries from the array, since blocks can be disconnected
+       * temporarily during mutation (which triggers this method).
+       */
+      syncArgsMap_: function (this: ProcedureBlock) {
+        // If we haven't yet stored the previous parameters, do so now. This would
+        // normally happen when we or initialize the procedure block with a model
+        // or update its parameters.
+        if (!this.prevParams_.length) {
+          this.prevParams_ = [...this.getProcedureModel().getParameters()];
```

Comment:
could `this.getProcedureModel()` technically return `null` here still if we don't find a procedure model?

Reply:
Technically, yes, it could, although I haven't found a way to hit that. This `syncArgsMap_` function is only called when parameters are updated, and the models should at least exist. Happy to add a null check. I think I can just update the spread part to `...(this.getProcedureModel()?.getParameters() || [])`

Reply:
yep might as well add the null check